app/services/atmosphere/optimizer.rb

```diff
@@ -14,6 +14,10 @@ def run(hint)
 
     #private
     def satisfy_appliance(appliance)
+      action = Action.create(appliance: appliance,
```

Comment:
Creating new entity about `Action` is performed by creating `Action` object and performing method `.log()`?